autoscale-workload-rabbit/src/main/java/com/github/autoscaler/workload/rabbit/RabbitSystemResourceMonitor.java

```diff
@@ -78,14 +90,30 @@ public ResourceUtilisation getCurrentResourceUtilisation() throws ScalerExceptio
                         }
                     }
                 }
-                memoryAllocated =  highestMemUsedInCluster;
-                diskFreeMbOpt = lowestDiskFreeMbInClusterOpt;
+                memoryAllocated = highestMemUsedInCluster;
+                rabbitDiskFreeMbOpt = lowestDiskFreeMbInClusterOpt;
+                if (config.getIsPayloadOffloadingEnabled()) {
+                    offloadingDiskFreeMbOpt = getOffloadingDiskFreeMb();
+                }
                 lastTime = System.currentTimeMillis();
             } catch (final IOException ex) {
                 throw new ScalerException("Unable to map response to status request.", ex);
             }
         }
-        return new ResourceUtilisation(memoryAllocated, diskFreeMbOpt);
+        return new ResourceUtilisation(memoryAllocated, rabbitDiskFreeMbOpt, offloadingDiskFreeMbOpt);
+    }
+
+    private Optional<Integer> getOffloadingDiskFreeMb() throws ScalerException
+    {
+        try {
+            final FileStore filestore = Files.getFileStore(Paths.get(config.getPayloadOffloadingDirectory()));
```

Comment:
There should be some validation here as someone could configure an empty string and Paths.get("") returns a Path representing the current working directory.